packages/flame/lib/src/components/component.dart

```diff
@@ -157,9 +155,15 @@ class Component with Loadable {
 
   @protected
   Vector2 eventPosition(PositionInfo info) {
-    return respectCamera
-        ? info.eventPosition.game
-        : info.eventPosition.viewportOnly;
+    if (positioningType == PositioningType.game) {
```

Comment:
A switch case should be safer here :)

Reply:
@luanpotter was complaining that switch in Dart wasn't exhaustive, but I tried now and it looks like it works fine. (If a `PositioningType` enum is added the type checker complains, which is what we want.)

Reply:
I think that if you don't specify the lint rule, there is no warning, but I think that we have that rule enabled